src/cobalt.rs

```diff
@@ -125,6 +168,43 @@ pub fn build(config: &Config) -> Result<()> {
         }
     }
 
+    // load data files
+    let mut data_map: HashMap<String, Value> = HashMap::new();
+    let data_root = source.join(&config.data);
+    let data_files_builder = FilesBuilder::new(data_root.as_path())?;
+    let data_files = data_files_builder.build()?;
+
+    for df in data_files.files() {
+
+        if let Some(ext) = df.extension() {
+            if let Some(file_stem) = df.file_stem() {
```

Comment:
Is there ever a time we'll get an [empty file_stem](https://doc.rust-lang.org/std/path/struct.Path.html#method.file_stem) from `.files()`?

Could we clean this up by doing
```rust
let file_stem = df.file_stem().expect("Files will always return with a stem");
```